modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBFilter.java

```diff
@@ -225,6 +225,25 @@ public MBFilter(JSONArray json, MBObjectParser parse, SemanticType semanticType)
         return Collections.emptySet();
     }
 
+    private Filter translateType(String jsonText) {
+        final FilterFactory2 ff = parse.getFilterFactory();
+        //TODO: How to wildcard geometry
+        Expression dimension = ff.function("dimension", ff.function("geometry"));
```

Comment:
There may be an argument for using the `geometryType` function instead of the `dimension` function here. Dimension is simpler, but I am not certain it captures all cases - in the case of time or elevation data, does dimension still map to the correct geometry type?